provenance.jq

```diff
@@ -0,0 +1,76 @@
+# input: "build" object (with "buildId" top level key)
+# output: list of image tags
+def tags:
+	[
+		.source.arches[].tags[],
+		.source.arches[].archTags[],
+		.build.img
+	]
+;
+
+# input: "tags" object with image digest and platform arguments
+# output: json object for in-toto provenance subject field
+def subjects($platform; $digest):
+	($digest | split(":")) as $splitDigest |
+	{
+		"name": ("pkg:docker/" + . + "?platform=" + $platform),
+		"digest": {
+			($splitDigest[0]): $splitDigest[1]
+		}
+	}
+;
+
+# input: GITHUB context argument
+# output: json object for in-toto provenance external parameters field
+def github_external_parameters($github):
+	($github.workflow_ref | gsub( $github.repository + "/"; "")) as $workflowPathRef |
+	{
+		inputs: $github.event.inputs,
+		workflow: {
+			ref: ($workflowPathRef | split("@")[1]),
+			repository: ($github.server_url + "/" + $github.repository),
+			path: ($workflowPathRef | split("@")[0]),
+			digest: {sha256: $github.workflow_sha}
```

Comment:
`workflow_sha` is just a commit digest, right?  so this should be either `sha1` or better yet `gitCommit` again?

```suggestion
			digest: { gitCommit: $github.workflow_sha },
```

(doesn't this actually belong in the "resolved dependencies" section?)

Reply:
yes, thank you

> (doesn't this actually belong in the "resolved dependencies" section?)

this was an addition I made to record the commit sha of the workflow (since it wasn't anywhere else) and the other fields here identify the workflow as part of https://github.com/actions/buildtypes/tree/main/workflow/v1

to me it was missing the commit, so I added it. In our use-case it is a bit redundant since the `meta` resolvedDependency == workflow commit